### PR TITLE
Add python bindings for Hydro lib

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

This was written by @aasthabagree21 a year or so ago. Trying to get this merged since it would be nice to have.

Depends on
- [x] https://github.com/sxs-collaboration/spectre/pull/6185

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/PointwiseFunctions/Hydro/Python/Bindings.cpp [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+  bind_stress_energy_impl<DataType>(m);
+}
+
+}  // namespace
+
+PYBIND11_MODULE(_Pybindings, m) {  // NOLINT
+  enable_segfault_handler();
+  py::module_::import("spectre.DataStructures");
+  py::module_::import("spectre.Spectral");
+  bind_impl<double>(m);
```
Comment (nilsvu) [inline src/PointwiseFunctions/Hydro/Python/Bindings.cpp]: Delete this `double` instantiation, we don't need on the Python side

File: src/PointwiseFunctions/Hydro/Python/Bindings.cpp [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+  bind_impl<DataVector>(m);
+  m.def("LowerSpatialFourVelocityCompute",
```
Comment (nilsvu) [inline src/PointwiseFunctions/Hydro/Python/Bindings.cpp]: Name `lower_spatial_four_velocity`

**Review by nilsvu — CHANGES_REQUESTED** [review]
Please rebase and squash

**Comment by nilsdeppe** [comment]
Rebased and squashed in fixes. Thank you! 🍻

**Review by nilsvu — APPROVED** [review]

**Pull request merged**